### DISCUSSION TITLE
Adding plots/histograms as overlays above a 3D scene

### DISCUSSION THREAD

**Discussion by tamasgal** [JuliaPlots/Makie.jl#3045] [Q&A]
I spent the night playing around with different constructs but I could not manage to display any additional scenes/plots on top of my animated 3D scene, so I hope to get tossed in the right direction here.

Below is a MWE which is a stripped down version of my main GL loop and a screenshot of it annotated with an example how I'd like to add overlays.

I am now trying to add graphical overlays (mostly plots/histograms and additional text, hopefully other GUI elements like buttons/dropboxes in future) but can't get them displayed at all.
At this point, I am not even sure if it's clever to layer the scenes like this, and also looked at the [Layout Tutorial](https://docs.makie.org/stable/tutorials/layout-tutorial/) but I guess that [Subwindows](https://docs.makie.org/stable/tutorials/scenes/#scenes_and_subwindows) are what I am looking for. However, the tutorial is a bit hard to grasp -- at least for me -- so apologies if this is trivial ;) I am definitely going to turn my tiny project into a tinier example and provide a PR to Makie, it's really great stuff so far!

Anyways, here is the MWE and the four lines below the comment are some desperate attempts to create a subscene and add some content, which is unfortunately not appearing anywhere on screen. I very likely miss to render them in the `while` loop but can't figure out how.

```julia
using GLMakie
using GLFW


function basegrid!(scene; center=(0, 0, 0), span=(-500, 500), spacing=50, linewidth=1, color=(:grey, 0.5))
    min, max = span
    center = Point3f(center)
    for q ∈ range(min, max; step=spacing)
        lines!(scene, [Point3f(q, min, 0) + center, Point3f(q, max, 0) + center], color=color, linewidth=linewidth)
        lines!(scene, [Point3f(min, q, 0) + center, Point3f(max, q, 0) + center], color=color, linewidht=linewidth)
    end
    scene
end


function main()
    println("Creating scene.")
    scene = Scene(backgroundcolor=RGBf(0.9))

    cam = cam3d!(scene, rotation_center = :lookat) # leave out if you implement your own camera

    lookat = Point3f(200, 200, 0)
    basegrid!(scene; center=lookat)

    center!(scene)
    update_cam!(scene, cam, Vec3f(1000), lookat)

    screen = display(GLMakie.Screen(start_renderloop=false), scene)

    pix = Makie.campixel(scene)
    frame_idx = 0
    framecounter = text!(pix, Point2f(10, 10), text = "t = 0 ns")

    # How to make this render on top of the scene (as overlay) in the loop below?
    subwindow = Scene(scene, px_area=Rect(100, 100, 200, 200), clear=true, backgroundcolor=:green)
    meshscatter!(subwindow, rand(Point3f, 10), color=:gray)
    plot!(subwindow, [1,2,3], rand(3))
    center!(subwindow)

    while isopen(screen)
        rotate_cam!(scene, Vec3f(0, 0.001, 0))

        framecounter.text = "t = $frame_idx ns"

        GLMakie.pollevents(screen)
        GLMakie.render_frame(screen)

        GLFW.SwapBuffers(GLMakie.to_native(screen))

        frame_idx += 1
    end
    GLMakie.destroy!(screen)
end

main()
```

![Screenshot 2023-07-05 at 08 28 54](https://github.com/MakieOrg/Makie.jl/assets/1730350/0e465f4e-70d5-4796-9e8f-860b7ef9da02)

**Comment by SimonDanisch** [comment]
Oh man, this only doesn't work, because the `px_area` isn't an observable...
```julia
    subwindow = Scene(scene, px_area=Observable(Rect(100, 100, 200, 200)), clear=true, backgroundcolor=:green)
    subwindow.clear = true
    meshscatter!(subwindow, rand(Point3f, 10), color=:gray)
    plot!(subwindow, [1, 2, 3], rand(3))
```
Definitely open an issue about that, or better a PR :)

**Reply by tamasgal** [reply]
Ah OK, I am not sure I understand the rendering though 😆 

So `clear=true` should clear everything behind the scene (I thought). `backgroundcolor=:green` sets the background to (wait for it) green.

However, this is what I see, which is kind of interesting since I have no clue where the spheres are coming from, nor where to see the `plot` or `meshscatter`. I think I am confused about the coordinate systems 😉

![Screenshot 2023-07-05 at 23 21 09](https://github.com/MakieOrg/Makie.jl/assets/1730350/b26f7039-7bf2-4f7a-8c0b-4ec6f334b530)
